File: dashboard/src/types/overview.ts
import type { Alert } from "@/types/alerts"

export type SavingsTrendPoint = {
  month: string
  savings: number
}

export type FindingTypeCount = {
  type: string
  count: number
}

export type LatestRunSummary = {
  runId: string
  status: string
  duration: string
}

export type Overview = {
  totalEstimatedMonthlySavings: number
  ec2FindingsCount: number
  s3FindingsCount: number
  blockedRiskyRecommendationsCount: number
  latestRun: LatestRunSummary
  savingsTrend: SavingsTrendPoint[]
  findingsByType: FindingTypeCount[]
  recentAlerts: Alert[]
}

File: dashboard_backend/app/schemas/overview.py
from pydantic import BaseModel

from .alerts import AlertResponse


class SavingsTrendPointResponse(BaseModel):
    month: str
    savings: float


class FindingTypeCountResponse(BaseModel):
    type: str
    count: int


class LatestRunSummaryResponse(BaseModel):
    runId: str
    status: str
    duration: str


class OverviewResponse(BaseModel):
    totalEstimatedMonthlySavings: float
    ec2FindingsCount: int
    s3FindingsCount: int
    blockedRiskyRecommendationsCount: int
    latestRun: LatestRunSummaryResponse
    savingsTrend: list[SavingsTrendPointResponse]
    findingsByType: list[FindingTypeCountResponse]
    recentAlerts: list[AlertResponse]
